src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseDataFileService.cs

```diff
@@ -428,89 +432,144 @@ public async Task<Either<ActionResult, DataFileInfo>> UploadAsZip(Guid releaseVe
                 });
         }
 
-        public async Task<Either<ActionResult, List<DataFileInfo>>> UploadAsBulkZip(
+        public async Task<Either<ActionResult, List<ArchiveDataSetFile>>> ValidateAndUploadBulkZip(
             Guid releaseVersionId,
-            IFormFile bulkZipFormFile)
+            IFormFile zipFile,
+            CancellationToken cancellationToken)
         {
             return await _persistenceHelper
                 .CheckEntityExists<ReleaseVersion>(releaseVersionId)
                 .OnSuccess(_userService.CheckCanUpdateReleaseVersion)
-                .OnSuccess(_ => _dataArchiveValidationService.ValidateBulkDataArchiveFile(
-                        releaseVersionId, bulkZipFormFile)
-                .OnSuccess(async dataArchiveFiles =>
+                .OnSuccessVoid(async _ =>
                 {
-                    var bulkZipFile = new File
+                    var errors = await _dataArchiveValidationService.IsValidZipFile(zipFile);
+                    if (errors.Count > 0)
                     {
-                        CreatedById = _userService.GetUserId(),
-                        RootPath = releaseVersionId,
-                        ContentLength = bulkZipFormFile.Length,
-                        ContentType = bulkZipFormFile.ContentType,
-                        Filename = bulkZipFormFile.FileName.ToLower(),
-                        Type = BulkDataZip,
-                    };
-                    _contentDbContext.Files.Add(bulkZipFile);
-                    await _contentDbContext.SaveChangesAsync();
+                        // TODO: Check this returns
+                        Common.Validators.ValidationUtils.ValidationResult(errors);
+                    }
+                })
+                .OnSuccess(async _ => await _dataArchiveValidationService.ValidateBulkDataArchiveFiles(releaseVersionId, zipFile))
+                .OnSuccess(async dataSetFiles =>
+                {
+                    await using var stream = zipFile.OpenReadStream();
+                    var archive = new ZipArchive(stream);
 
-                    // each data/meta file pair are extracted to blob storage by _dataImportService.Import
-                    await UploadFileToStorage(bulkZipFile, bulkZipFormFile);
+                    foreach (var file in dataSetFiles)
+                    {
+                        file.DataFilePath = $"{FileStoragePathUtils.FilesPath(releaseVersionId, FileType.Data)}{Guid.NewGuid()}";
+                        var dataFileEntry = archive.GetEntry(file.DataFilename.ToLower());
```

Comment:
I think I'd remove the `ToLower` here. With it, I think with it any original file containing capital letters won't be found.

Same when fetching `metaFileEntry` below.